tests/integration/customizations/test_override_ssl_common_name.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from awscli.testutils import mock, aws
+import pytest
+
+from awscli.customizations.overridesslcommonname import ssl_common_names
+
+regions_to_exclude = [
+    "af-south-1",
+    "ap-east-1",
+    "ap-southeast-3",
+    "eu-south-1",
+    "me-south-1",
+    "cn-north-1",
+    "cn-northwest-1",
+    "us-gov-west-1",
+    "us-isob-east-1",
+]
+sqs = [
+    ("sqs", "list-queues", region)
+    for region in ssl_common_names["sqs"]
+    if region not in regions_to_exclude
+]
+emr = [
+    ("emr", "list-clusters", region)
+    for region in ssl_common_names["emr"]
+    if region not in regions_to_exclude
+]
+shield = [
+    ("shield", "list-protections", region) for region in ssl_common_names["shield"]
+]
+rds = [("rds", "describe-db-clusters", region) for region in ssl_common_names["rds"]]
+test_cases = sqs + emr + shield + rds
+
+
+
+@pytest.mark.parametrize("service,operation,region", test_cases)
+def test_update_endpoint_url(service, operation, region):
+
+    p = aws(f"{service} {operation} --region {region}")
+    assert p.rc == 0
+    if service == 'emr':
+        assert p.json == {'Clusters': []}
+    if service == 'shield':
+        assert p.json == {'Protections': []}
+    if service == 'rds':
+        assert p.json == {'DBClusters': []}
```

Comment:
This has a success case where we end up with an unknown service and don't make any assertions. Those kind of edge cases can lead to the test subtly changing, resulting in nothing being checked. We're also not asserting anything about SQS.

I also don't know if this will pass in an actual account. As soon as any resources are added, these tests will fail. I'd probably bias against doing integration tests for these, but if we decide to, let's make a more general assertion that we got a valid JSON response and it was not empty.

Reply:
I omitted SQS because I was getting back an empty string in the `stdout` which won't translate to valid json. i'll see if i can find another op that returns valid json even if there are no results.

Maybe I can add the keys to the test case tuples and check that its in the json?

Reply:
there also don't seem to be any other operations for sqs that don't require an existing resources besides ones that create them. i could use one of those and add a cleanup function at the end